src/test/StdCheats.t.sol

```diff
@@ -76,6 +76,48 @@ contract StdCheatsTest is Test {
         assertEq(barToken.totalSupply(), 10000e18);
     }
 
+    function testBound() public {
+        assertEq(bound(5, 0, 4), 0);
+        assertEq(bound(0, 69, 69), 69);
+        assertEq(bound(0, 68, 69), 68);
+        assertEq(bound(10, 150, 190), 160);
+        assertEq(bound(300, 2800, 3200), 3100);
+        assertEq(bound(9999, 1337, 6666), 6006);
+        
+    }
+
+    function testCannotBoundMaxLessThanMin() public {
+        vm.expectRevert(bytes("MAX_LESS_THAN_MIN"));
+        bound(5, 100, 10);
+    }
+
+    function testBound(
+        uint256 num,
+        uint256 min,
+        uint256 max
+    ) public {
+        if (min > max) (min, max) = (max, min);
+
+        uint256 bounded = bound(num, min, max);
+
+        assertGe(bounded, min);
+        assertLe(bounded, max);
+    }
+
+    function testBoundUint256Max() public {
+        assertEq(bound(0, type(uint256).max - 1, type(uint256).max), type(uint256).max - 1);
```

Comment:
We should also to test the case when the `result` is `type(uint256).max`.
```suggestion
        assertEq(bound(0, type(uint256).max - 1, type(uint256).max), type(uint256).max - 1);
        assertEq(bound(1, type(uint256).max - 1, type(uint256).max), type(uint256).max);
```

Reply:
done :)